### PR TITLE
chore: bump iminuit to v2.29.1, and `scikit-learn` to v1.5.2

### PR DESCRIPTION
**Pull request by agriyakhetarpal**
### Description

This PR bumps two packages:
- `iminuit` to [version `2.29.1`](https://github.com/scikit-hep/iminuit/releases/tag/v2.29.1). The patch previously added in #4767 can now be deleted and dependence on `packaging` can be removed. Tagging the recipe maintainer @henryiii for visibility.
- `scikit-learn` to [version `1.5.2`](https://github.com/scikit-learn/scikit-learn/releases/tag/1.5.2). Tagging the recipe maintainer @lesteve for visibility.

I was previously involved with helping review the files associated with their recipes, so I thought it would be good to get updated versions of them before 0.27 stable lands.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [ ] Add / update tests

### CODE REVIEW

**Comment by lesteve** [comment]
Thanks for this! Out of curiosity, do you have a rough timeline for Pyodide 0.27?

**Comment by agriyakhetarpal** [comment]
> Thanks for this! Out of curiosity, do you have a rough timeline for Pyodide 0.27?

@lesteve oops, forgot to link it in the description, I thought I had done that: #5064

**Comment by agriyakhetarpal** [comment]
All relevant tests for `iminuit` and `scikit-learn` passing, we can merge this.

File: packages/iminuit/meta.yaml [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
 test:
   imports:
     - iminuit
 requirements:
   run:
     - numpy # runtime only
-    - packaging
 build:
   cxxflags: -fexceptions
   ldflags: -fexceptions
```
Comment (henryiii) [inline packages/iminuit/meta.yaml:15]: These also can be removed, I believe.

**Pull request merged**